domain/transaction.go

```diff
@@ -6,12 +6,32 @@ import (
 	"github.com/dogmatiq/example/messages/events"
 )
 
+// transaction is the aggregate root for a bank transaction.
+type transaction struct {
+	// Recorded is the recorded state of the transaction.
+	Recorded bool
```

Comment:
```suggestion
	Started bool
```

Perhaps?

Reply:
Its not something that stops though?

Reply:
All of the events are name like `DepositStarted`, etc. Not `DepositRecorded`, what is recorded?

Reply:
Recorded as it there is a record of this having occurred. All good though, i'll change to Started.